patches/minecraft/net/minecraft/advancements/AdvancementRewards.java.patch

```diff
@@ -9,3 +32,55 @@
        boolean flag = false;
  
        for(ResourceLocation resourcelocation : this.f_9980_) {
+@@ -69,6 +_,7 @@
+          p_9990_.m_7902_(this.f_9981_);
+       }
+ 
++      net.minecraftforge.common.advancements.CustomAdvancementRewardHooks.grant(p_9990_, this.customRewards);
+       MinecraftServer minecraftserver = p_9990_.f_8924_;
+       this.f_9982_.m_78002_(minecraftserver.m_129890_()).ifPresent((p_9996_) -> {
+          minecraftserver.m_129890_().m_136112_(p_9996_, p_9990_.m_20203_().m_81324_().m_81325_(2));
+@@ -108,6 +_,7 @@
+             jsonobject.add("recipes", jsonarray1);
+          }
+ 
++         net.minecraftforge.common.advancements.CustomAdvancementRewardHooks.serialize(jsonobject, this.customRewards);
+          if (this.f_9982_.m_77999_() != null) {
+             jsonobject.addProperty("function", this.f_9982_.m_77999_().toString());
+          }
+@@ -139,13 +_,15 @@
+          commandfunction$cacheablefunction = CommandFunction.CacheableFunction.f_77990_;
+       }
+ 
+-      return new AdvancementRewards(i, aresourcelocation, aresourcelocation1, commandfunction$cacheablefunction);
++      net.minecraftforge.common.advancements.ICustomAdvancementReward[] customRewards = net.minecraftforge.common.advancements.CustomAdvancementRewardHooks.deserialize(p_9992_);
++      return new AdvancementRewards(i, aresourcelocation, aresourcelocation1, commandfunction$cacheablefunction, customRewards);
+    }
+ 
+    public static class Builder {
+       private int f_9999_;
+       private final List<ResourceLocation> f_10000_ = Lists.newArrayList();
+       private final List<ResourceLocation> f_10001_ = Lists.newArrayList();
++      private final List<net.minecraftforge.common.advancements.ICustomAdvancementReward> customRewards = Lists.newArrayList();
```

Comment:
`Lists#newArrayList` is considered deprecated by guava as of Java 7+. Use the `ArrayList` constructor instead.